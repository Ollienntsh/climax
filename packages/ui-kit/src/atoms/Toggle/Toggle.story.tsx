import React from 'react';
import { storiesOf } from '@storybook/react';

import Toggle from './Toggle';
import Flex from '../Flex';

storiesOf('ui-kit/Atoms', module).add('Toggle', () => (
  <Flex gutter={2} padding={20}>
    <span>Toggle:</span>
    <Toggle />
  </Flex>
));
