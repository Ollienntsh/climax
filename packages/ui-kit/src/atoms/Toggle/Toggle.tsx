import { Switch as AntSwitch } from 'antd';
import styled from '@emotion/styled';

export default styled(AntSwitch)({});
