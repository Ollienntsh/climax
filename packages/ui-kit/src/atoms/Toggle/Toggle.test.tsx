import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import snapshotDiff from 'snapshot-diff';

import Toggle from './Toggle';

expect.addSnapshotSerializer(serializer);

const defaultTree = renderer.create(<Toggle />).toJSON();

describe('Toggle', () => {
  it('should render default', () => {
    expect(defaultTree).toMatchSnapshot();
  });
  it('should render Toggle active', () => {
    const tree = renderer.create(<Toggle defaultChecked />).toJSON();

    expect(snapshotDiff(tree, defaultTree)).toMatchSnapshot();
  });
});
