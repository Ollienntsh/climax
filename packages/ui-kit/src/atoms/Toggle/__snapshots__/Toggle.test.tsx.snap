// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render Toggle active 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  <button
-   aria-checked={true}
-   className=\\"css-0 ant-switch ant-switch-checked\\"
-   defaultChecked={true}
+   aria-checked={false}
+   className=\\"css-0 ant-switch\\"
+   defaultChecked={false}
    onClick={[Function handleClick]}
    onKeyDown={[Function handleKeyDown]}
    onMouseUp={[Function handleMouseUp]}
    role=\\"switch\\"
    type=\\"button\\""
`;

exports[`Toggle should render default 1`] = `
<button
  aria-checked={false}
  className="css-0 ant-switch"
  defaultChecked={false}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseUp={[Function]}
  role="switch"
  type="button"
>
  <span
    className="ant-switch-inner"
  />
</button>
`;
